lib/omniauth/strategies/openid_connect.rb

```diff
@@ -294,6 +297,18 @@ def logout_path_pattern
         @logout_path_pattern ||= %r{\A#{Regexp.quote(request_path)}(/logout)}
       end
 
+      def validate_grant_type_credentials
+        # It supports the combined 'id_token token' response type
+        options.response_type.to_s.split(' ').each do |grant_type|
+          grant_type_param = grant_type == 'token' ? 'access_token' : grant_type
+          unless params.key?(grant_type_param) && SUPPORTED_GRANT_TYPES.include?(grant_type)
+            error_class = "OmniAuth::OpenIDConnect::Missing#{grant_type.classify}Error".constantize
```

Comment:
Avoid using Rails methods like `#constantize`

Reply:
I actually checked if we use activesupport and it seems that we do, and in such a case it's fine to use `constantize` in my opinion.

`openid_connect` has `active_model` as a runtime dependency, and `active_model` depends on `active_support` (which defines `constantize`).

Reply:
So you rely on the fact that we have a dependency, which has dependency, which has another dependency.

No, sorry, but this is not fine.

More of that, ActiveSupport does not provide any methods that could be really useful here. `#present` is useful when you don't want to cast object, and can check empty string, `nil`, `false`, etc.. None of that is really need here since you check the result of your own method.

Same with `#constantize`: there's no real reason to dynamically create a constant:

```ruby
CLASSES_MAP = {
  'code' => OmniAuth::OpenIDConnect::MissingCodeError,
  'id_token' => OmniAuth::OpenIDConnect::MissingIdTokenError,
}.freeze

error_class = CLASSES_MAP.fetch(grant_type)
```

Reply:
Also, the dynamic code is always a bad thing for refactoring. Consider this too.

Reply:
The fact is that this gem uses ActiveSupport.
If using these 2 methods are blockers for you, I can remove them.

Reply:
1) We can not rely on the fact that we have a dependency indirectly. If you want to use the library you have to add it to dependencies explicitly.
2) There's no real reason to use these methods. I have shown why and how to get rid of them.

These two facts make using these methods inappropriate to me, yes. It would be great if you can get rid of them.